### PR TITLE
docker: url-encode username and password

### LINKED ISSUE #60: Slashes (`/`) in redis password break URI generation
If the redis password contains a forward slash (`/`), an invalid URI is generated:

```
/usr/local/lib/ruby/3.2.0/uri/rfc3986_parser.rb:66:in `split': bad URI(is not URI?): "redis://redis:passwordwitha/slash@redishostname:6379/0" (URI::InvalidURIError)
    from /usr/local/lib/ruby/3.2.0/uri/rfc3986_parser.rb:71:in `parse'
    from /usr/local/lib/ruby/3.2.0/uri/common.rb:193:in `parse'
    from /usr/local/bundle/gems/sidekiq-6.5.8/lib/sidekiq/redis_connection.rb:133:in `log_info'
    from /usr/local/bundle/gems/sidekiq-6.5.8/lib/sidekiq/redis_connection.rb:103:in `create'
    from /usr/local/bundle/gems/sidekiq-6.5.8/lib/sidekiq.rb:205:in `redis='
    from /app/config.ru:42:in `block (2 levels) in <main>'
    from /usr/local/bundle/gems/sidekiq-6.5.8/lib/sidekiq.rb:152:in `configure_client'
    from /app/config.ru:42:in `block in <main>'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/builder.rb:116:in `eval'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/builder.rb:116:in `new_from_string'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/builder.rb:105:in `load_file'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/builder.rb:66:in `parse_file'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/server.rb:349:in `build_app_and_options_from_config'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/server.rb:249:in `app'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/server.rb:422:in `wrapped_app'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/server.rb:312:in `block in start'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/server.rb:379:in `handle_profiling'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/server.rb:311:in `start'
    from /usr/local/bundle/gems/rack-2.2.7/lib/rack/server.rb:168:in `start'
    from /usr/local/bundle/gems/rack-2.2.7/bin/rackup:5:in `<top (required)>'
    from /usr/local/bundle/bin/rackup:25:in `load'
    from /usr/local/bundle/bin/rackup:25:in `<main>'
```

### PR DESCRIPTION
**Pull request by roobre**
Fixes #60

### CODE REVIEW

File: docker/config.ru [hunk 1]
```diff
@@ -12,8 +12,8 @@ unless config.key?(:url)
   port = ENV.fetch('REDIS_PORT', 6379)
   db_number = ENV.fetch('REDIS_DB_NUMBER', 0)
 
-  credentials = ENV.fetch('REDIS_USERNAME', '')
-  credentials = "#{credentials}:#{ENV['REDIS_PASSWORD']}" if ENV.key?('REDIS_PASSWORD')
+  credentials = ERB::Util.url_encode(ENV.fetch('REDIS_USERNAME', ''))
```
Comment (Strech) [inline docker/config.ru:15]: Thanks a lot for fixing the bug.

I would like to ask you to use `URI::DEFAULT_PARSER.escape` to escape url dangerous characters.

**Review by Strech — CHANGES_REQUESTED** [review]
Thanks a lot for the fix. I would like to suggest more relevant standard lib class for fix it

File: docker/config.ru (hunk 1 above)
Comment (roobre) [reply docker/config.ru]: Done!

**Comment by Strech** [comment]
Thanks a lot for your help, latest version of the Docker image along side with Helm package were released as [`0.2.0-2`](https://github.com/Strech/sidekiq-prometheus-exporter/releases/tag/v0.2.0-2) 🎉

**Pull request merged**